smarts/core/trap_manager.py

```diff
@@ -193,15 +188,18 @@ def largest_vehicle_plane_dimension(vehicle):
                 )
             elif trap.patience_expired:
                 mission = trap.mission
-                if len(agent_vehicle_comp) > 0:
-                    agent_vehicle_comp.sort(
+                if len(vehicle_comp) > 0:
+                    vehicle_comp.sort(
                         key=lambda v: squared_dist(v[0], mission.start.position)
                     )
 
-                    # Make sure there is not an agent vehicle in the same location
-                    pos, largest_dimension, _ = agent_vehicle_comp[0]
-                    if squared_dist(pos, mission.start.position) < largest_dimension:
-                        continue
+                    # Make sure there is not a vehicle in the same location
+                    for pos, largest_dimension, _ in vehicle_comp[:3]:
```

Comment:
There was a minor assumption that checking against the nearest 3 vehicles would be enough to see if any are intruding. A higher number (e.g. 4-6) is probably better if we are preempting issues with small vehicles like bikes.

Reply:
Alternatively, I don't think the performance hit would be that bad to just check them all.  What do you think?

Reply:
as discussed offline, just changed it to check all of them.  also, fixed a bug here with the continue scope!!